src/clarity-angular/datagrid/providers/sort.ts

```diff
@@ -48,15 +48,24 @@ export class Sort {
     };
 
     /**
-     * Sets a comparator as the current one, or toggles reverse if the comparator is already used.
+     * Sets a comparator as the current one, or toggles reverse if the comparator is already used. The
+     * optional forceReverse input parameter allows to override that toggling behavior by sorting in
+     * reverse order if `true`.
+     *
+     * @param {Comparator<any>} sortBy the comparator to use for sorting
+     * @param {boolean} [forceReverse] `true` to force sorting descendingly
+     *
+     * @memberof Sort
      */
-    public toggle(sortBy: Comparator<any>) {
+    public toggle(sortBy: Comparator<any>, forceReverse?: boolean) {
         // We modify private properties directly, to batch the change event
         if (this.comparator === sortBy) {
-            this._reverse = !this._reverse;
+            this._reverse = forceReverse !== undefined
```

Comment:
Please use `typeof forceReverse !== "undefined"`, it's safer on older browsers that let you reassign `undefined`. Not sure if we support any of these, but it doesn't cost much.